### ISSUE TITLE
Request for Conditional Statement for DHCP Server in Underlay templates

### ISSUE THREAD

**Issue by ramakrishnanr89** [fortinet-solutions-cse/sdwan-advpn-reference#3]
Could you please add a conditional statement for DHCP Server. Even when I do not want DHCP Server on the "lan" interface,  the program still enters the "config system dhcp server". Unlike creation of lan_zone where there is a condition to check the option before entering the "config system zone".

I am expecting something like this, {% if project.create_lan_dhcp_server|default(true) %} , with an option in project template " create_lan_dhcp_server".

Its trivial, but this will be useful :) 

Feel free to correct me, if I have misinterpreted the code.

**Comment by dmitryperets** [comment]
Hi @ramakrishnanr89, 

The difference with the LAN zone is that for the DHCP server we allow more granular, per-interface control. You can configure `dhcp_server: false` under your LAN interface in the profile, and then we won't configure a DHCP server for this particular LAN interface:

```
config system dhcp server
  {% for i in project.profiles[profile].interfaces if i.role == 'lan'  %}
  {% if i.dhcp_server|default(true) %}       <<<<<<<<<<<<<<<<<<<<<<<<<< here
  edit {{ 10 + loop.index0 }}
    ...
  next
  {% endif %}
  {% endfor %}
end
```

So the idea is to allow having some LAN interfaces with DHCP server and others - without. 
If you set the value to "false", the rendering will still generate the empty stanza, like: 

```
config system dhcp server
end
```

But that's nothing wrong, in our view... we could, of course, add a global option in addition to the selector per-interface.

**Comment by ramakrishnanr89** [comment]
Thanks. I created a global switch - create_lan_dhcp_server = false
to turn off dhcp-server. That way for customers who use Windows AD as DHCP (which is most India region customers), I can turn it off completely. That way it won't render the empty syntax as well.

I know its trivial, but just wanted to highlight in case someone else also was trying to figure this out.

**Comment by dmitryperets** [comment]
Yeah, that's a good idea. I'll add the global switch too. Thanks!

**Comment by dmitryperets** [comment]
#4 merged into `release/7.2`
